frame/support/procedural/src/construct_runtime/parse.rs

```diff
@@ -198,6 +198,62 @@ impl Parse for PalletDeclaration {
 	}
 }
 
+/// A struct representing a path to a pallet. `PalletPath` is almost identical to the standard
+/// Rust path with a few restrictions:
+/// - No leading colons allowed
+/// - Path segments can only consist of identifers; angle-bracketed or parenthesized segments will
+///   result in a parsing error (except when specifying instances)
+#[derive(Debug, Clone)]
+pub struct PalletPath {
+	pub inner: Path,
+}
+
+impl Parse for PalletPath {
+	fn parse(input: ParseStream) -> Result<Self> {
+		let mut lookahead = input.lookahead1();
+		let mut segments = Punctuated::new();
+
+		while lookahead.peek(Ident) {
```

Comment:
I think I should technically also allow for keywords like `self` and `crate`  as the first path segment.

Reply:
Weird, `PathSegment` doesn't actually allow `self`, `super` or `crate` keywords to appear. I've no idea how it manages to parse paths like `self::foo::bar`...

Reply:
Nevermind, solved with `Ident::parse_any`.